src/functions/severa/get-filtered-workhours/handler.ts

```diff
@@ -0,0 +1,95 @@
+import type { APIGatewayProxyHandler } from "aws-lambda";
+import { CreateSeveraApiService } from "src/database/services/severa-api-service";
+import { middyfy } from "src/libs/lambda";
+import { FilterUtilities } from "src/libs/filter-utils";
+import type WorkHours from "src/types/severa/workHour/workHour";
+import type SeveraResponse from "src/types/severa/workHour/severaResponseWorkHours";
+
+/**
+ * Handler for getting work hours from Severa REST API.
+ *
+ * @param event - API Gateway event containing the userId, projectId, and phaseId.
+ */
+export const getWorkHoursHandler: APIGatewayProxyHandler = async (event) => {
+  const { severaUserId, severaProjectId, severaPhaseId } = event.pathParameters || {};
+  const { startDate, endDate } = event.queryStringParameters || {};
+
+  try {
+    const api = CreateSeveraApiService();
+
+    const buildWorkHoursUrl = (severaProjectId?: string, severaUserId?: string, startDate?: string, endDate?: string) => {
+      let url: string;
+
+      if (severaProjectId) {
+        url = `projects/${severaProjectId}/workhours`;
+      } else if (severaUserId) {
+        url = `users/${severaUserId}/workhours`;
+      } else {
+        url = "workhours";
+      }
+    
+      const queryParams = new URLSearchParams();
+    
+      if (startDate) queryParams.append("startDate", startDate);
+      if (endDate) queryParams.append("endDate", endDate);
+    
+      return queryParams.toString() ? `${url}?${queryParams.toString()}` : url;
+    };
+
+    const url = buildWorkHoursUrl(severaProjectId, severaUserId, startDate, endDate);
+    
+    const response = await api.getWorkHoursBySeveraId(url);
+
+    const filteredWorkHours = response.filter((workHours:SeveraResponse) => {
+      if (severaProjectId && severaUserId) {
+        const filteredWorkHoursUserProject = FilterUtilities.filterByUserSevera(workHours.user?.guid, severaUserId);
+        if (!filteredWorkHoursUserProject) return false ;
+      }
```

Comment:
Might be worth adding a comment here to make it clear that in the case of there being severaProjectId and severaUserId it will have been filtered by project from the severaApi (due to the building of workhours url).